precompiles/parachain-staking/src/lib.rs

```diff
@@ -180,6 +190,105 @@ where
 		})
 	}
 
+	fn candidate_count(target_gas: Option<u64>) -> Result<PrecompileOutput, ExitError> {
+		let mut gasometer = Gasometer::new(target_gas);
+
+		// Fetch info.
+		gasometer.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let candidate_count: u32 = <parachain_staking::Pallet<Runtime>>::candidate_pool()
+			.0
+			.len() as u32;
+
+		// Build output.
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			cost: gasometer.used_gas(),
+			output: EvmDataWriter::new().write(candidate_count).build(),
+			logs: vec![],
+		})
+	}
+
+	fn collator_nomination_count(
+		mut input: EvmDataReader,
+		target_gas: Option<u64>,
+	) -> Result<PrecompileOutput, ExitError> {
+		let mut gasometer = Gasometer::new(target_gas);
+
+		// Read input.
+		input.expect_arguments(1)?;
+		let address: Runtime::AccountId = input.read::<Address>()?.0.into();
+
+		// Fetch info.
+		gasometer.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
```

Comment:
It would be more readable to move this line just below the measured instruction.